pageserver/src/layered_repository.rs

```diff
@@ -844,7 +851,11 @@ impl Timeline for LayeredTimeline {
         let (seg, seg_blknum) = SegmentTag::from_blknum(rel, rel_blknum);
 
         if let Some((layer, lsn)) = self.get_layer_for_read(seg, lsn)? {
+            let tenant_id = self.tenantid.to_string();
+            let timeline_id = self.timelineid.to_string();
+
             RECONSTRUCT_TIME
+                .with_label_values(&[&tenant_id, &timeline_id])
```

Comment:
I'm afraid this could be very expensive, see https://github.com/zenithdb/zenith/issues/622. Can you see any measurable differences in our perf test suite? If not, perhaps we need to add a test case that would catch it.

Our solution to #622 was this: #638. I think we could do the same here: get a reference to the per-timeline Metric once, and hold it in LayeredTimeline. If we get more metrics like this, it would be good to group them together in the LayeredTimeline struct, or even add a new "TimelineMetrics" struct to hold them all.

Reply:
Fixed below with references in LayeredTimeline. I'll check perf test for any impact.

Reply:
Fixed below.